src/libraries/System.Resources.Extensions/src/System/Resources/Extensions/DeserializingResourceReader.cs

```diff
@@ -228,5 +248,17 @@ private object DeserializeObject(int typeIndex)
             return value;
         }
 
+        private static bool GetUseBinaryFormatterSetting()
+        {
+            if (AppContext.TryGetSwitch("System.Resources.Extensions.UseBinaryFormatter", out bool fileConfig))
+            {
+                return fileConfig;
+            }
+
+            string? envVar = Environment.GetEnvironmentVariable("DOTNET_SYSTEM_RESOURCES_EXTENSIONS_USE_BINARY_FORMATTER");
```

Comment:
I do not think that this should have env variable settings. This is not something we should be encouraging to enable machine wide.

Reply:
The binary formatter is going to throw by default in .NET 9. Apps will have to explicitly reference the OOB package to make it work again. It means that this env variable is effectively "break apps that were not rebuilt with explicitly reference to the OOB package".

If there are apps that need 100% compatible binary deserializer, asking them to set the config setting is not that much extra burden over adding the explicitly reference to the OOB package.